plugins/algoltek-aux/fu-algoltek-aux-device.c

```diff
@@ -0,0 +1,560 @@
+{	
+	g_autoptr(GByteArray) st = fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_new();
+	gsize length = st->len - 3;
+
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_i2c_address(st, 0x51);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_sublen(st, length + 1);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_len(st, length);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_cmd(st, FU_ALGOLTEK_AUX_CMD_BOT);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_address(st, address);
+
+	if(!fu_algoltek_dpaux_device_write(FU_DPAUX_DEVICE(self), st, 20, error))
+		return FALSE;	
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_algoltek_aux_device_ers(FuAlgoltekAuxDevice *self, GError **error)
+{	
+	g_autoptr(GByteArray) st = fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_new();
+	gsize length = st->len - 3;
+
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_i2c_address(st, 0x51);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_sublen(st, length + 1);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_len(st, length);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_cmd(st, FU_ALGOLTEK_AUX_CMD_ERS);
+	fu_struct_algoltek_aux_bot_ers_cmd_address_pkt_set_address(st, 0x6000);
+
+	if(!fu_algoltek_dpaux_device_write(FU_DPAUX_DEVICE(self), st, 20, error))
+		return FALSE;	
+
+	/* success */
+	return TRUE;
+}
+
+
+static GByteArray *
+fu_algoltek_aux_device_wrfcrc(FuAlgoltekAuxDevice *self, int serialNo, guint16* WCRC, GError **error)
+{
+	g_autoptr(GByteArray) st = fu_struct_algoltek_aux_crc_cmd_address_pkt_new();
+	gsize length = st->len - 3;
+    
+	fu_struct_algoltek_aux_crc_cmd_address_pkt_set_i2c_address(st, 0x51);
+	fu_struct_algoltek_aux_crc_cmd_address_pkt_set_sublen(st, length | 0x80);
+	fu_struct_algoltek_aux_crc_cmd_address_pkt_set_serialno(st, serialNo);
+	fu_struct_algoltek_aux_crc_cmd_address_pkt_set_len(st, 0x04);
+	fu_struct_algoltek_aux_crc_cmd_address_pkt_set_cmd(st, FU_ALGOLTEK_AUX_CMD_ISP);
+	fu_struct_algoltek_aux_crc_cmd_address_pkt_set_wcrc(st, *WCRC);
+	
+	if(!fu_algoltek_dpaux_device_write(FU_DPAUX_DEVICE(self), st, 10, error))
+		return FALSE;	
+
+	/* success */
+	return TRUE;
+
+}
+
+static gboolean
+fu_algoltek_aux_device_wrf(FuAlgoltekAuxDevice *self, GInputStream *stream, guint16* WCRC, FuProgress *progress, GError **error)
+{	
+	g_autoptr(GByteArray) st = fu_struct_algoltek_aux_isp_flash_write_cmd_address_pkt_new();
+	gsize length = st->len - 3;
+    guint16 serialNo = 1;
+    guint8 startLength = 0;
+	guint firmwareBufferCount = 0;
+
+	g_autoptr(FuChunkArray) chunks = NULL;
+	chunks = fu_chunk_array_new_from_stream(stream, 0, 8, error);
+	if (chunks == NULL)
+		return FALSE;
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_set_steps(progress, fu_chunk_array_length(chunks));
+
+	g_autoptr(GByteArray) firmwareBytesArray = NULL;
+	firmwareBytesArray = fu_input_stream_read_byte_array(stream, 0, AG_FIRMWARE_AUX_SIZE, error);
+	if (firmwareBytesArray == NULL) {
+		return FALSE;
+	} 
```

Comment:
no braces needed for a single line return

Reply:
Remove extra braces.